src/main/java/org/spongepowered/common/mixin/core/server/management/MixinPlayerInteractionManager.java

```diff
@@ -96,77 +110,78 @@ public void onBlockRemoving(final BlockPos pos, final CallbackInfo ci) {
     }
 
     /**
-     * @author gabizou - September 5th, 2018
-     * @reason Due to the way that buckets and the like can be handled
-     * on the client, often times we need to cancel the item stack usage
-     * due to server side cancellation logic that may not exist on the client.
-     * Therefor, the cancellation of possible block changes doesn't take
-     * effect, and therefor requires telling the client to set back the item
-     * in hand.
+     * @author morpheus - December 15th, 2018
      *
-     * @param actionResult The action result returned from useItemRightClick, which if
-     *  the result is FAIL, then we should be setting the item in hand back.
-     * @param player The player
-     * @param worldIn The world
-     * @param stack The stack
-     * @param hand The hand
-     * @return The result, but we will inject a "send inventory to player packet" fi it was failed
+     * @reason Fire interact block event.
      */
-    @Redirect(
-        method = "processRightClick",
-        at = @At(
-            value = "INVOKE",
-            target = "Lnet/minecraft/util/ActionResult;getType()Lnet/minecraft/util/EnumActionResult;",
-            ordinal = 0 // We need to target only the first getType, since
-                      // there's technically two getTypes that are being caught by the slice
-
-        ),
-        slice = @Slice(
-            from = @At(
-                value = "INVOKE",
-                target = "Lnet/minecraft/item/ItemStack;getMaxItemUseDuration()I",
-                ordinal = 0 // This targets the first max use duration in the massive if statement.
-            ),
-            to = @At(
-                value = "FIELD",
-                target = "Lnet/minecraft/util/EnumActionResult;FAIL:Lnet/minecraft/util/EnumActionResult;"
-            )
-        )
-    )
-    private EnumActionResult spongeGetResultCheckForFailure(ActionResult<ItemStack> actionResult, EntityPlayer player, net.minecraft.world.World worldIn, ItemStack stack, EnumHand hand) {
-        // Sanity checks on the world being used (hey, i don't know the rules about clients...
-        // and if the world is in fact a responsible server world.
-        final EnumActionResult result = actionResult.getType();
-        if (!(worldIn instanceof IMixinWorld) || ((IMixinWorld) worldIn).isFake()) {
-            return result;
+    @Overwrite
+    public void onBlockClicked(BlockPos pos, EnumFacing side) {
```

Comment:
If this is an overwrite, please comment existing method and why things are being done like so:
```java
@Overwrite
public void onBlockClicked(BlockPos pos, EnumFacing side) {
  // Sponge Start - throw a pre-event for block clicks to verify if they need to be cancelled
  final Blocksnapshot ........
  // Sponge End
  if (this.isCreative()) {
    vanilla code;
  }
}
```